xalia/Win32/AccessibleProvider.cs

```diff
@@ -243,6 +273,202 @@ public override bool WatchProperty(UiDomElement element, GudlExpression expressi
+        }
+
+        private List<ElementIdentifier> GetChildrenEnumVariantBackground()
+        {
+            var count = IAccessible.accChildCount;
+
+            return GetChildrenEnumVariantBackground(count);
+        }
+
+        unsafe private List<ElementIdentifier> GetChildrenEnumVariantBackground(int count)
+        {
+            if (count == 0)
+                return new List<ElementIdentifier>(0);
+
+            IEnumVARIANT e;
+            try
+            {
+                e = (IEnumVARIANT)IAccessible;
+            }
+            catch (InvalidCastException)
+            {
+                return null;
+            }
+
+            int res = e.Reset();
+            Marshal.ThrowExceptionForHR(res);
+
+            object[] variants = new object[count];
+            res = e.Next(count, variants, new IntPtr(&count));
+            Marshal.ThrowExceptionForHR(res);
+
+            List<ElementIdentifier> result = new List<ElementIdentifier>(count);
+            HashSet<object> seen = new HashSet<object>(count);
+            foreach (var v in variants)
+            {
+                if (v is int i && i == CHILDID_SELF)
+                    continue;
+                if (!seen.Add(v))
+                    continue;
+                result.Add(ElementIdFromVariantBackground(v));
+            }
+
+            return result;
+        }
+
+        private ElementIdentifier ElementIdFromVariantBackground(object variant)
+        {
+            ElementIdentifier result = default;
+            result.root_hwnd = RootHwnd.Hwnd;
+            IAccessible acc;
+            if (variant is int childid)
+            {
+                var child = IAccessible.accChild[childid];
+                if (child is null)
+                {
+                    // Child without its own IAccessible
+                    result.acc = IAccessible;
+                    result.child_id = childid;
+                    return result;
+                }
+                acc = (IAccessible)child;
+            }
+            else if (variant is IAccessible ia)
+                acc = ia;
+            else if (variant is null)
+                throw new InvalidOperationException($"variant is VT_EMPTY");
+            else
+                throw new InvalidOperationException($"wrong type {variant.GetType()}");
+
+            IOleWindow oleWindow = acc as IOleWindow;
+            if (!(oleWindow is null))
+            {
+                try
+                {
+                    IntPtr root_hwnd = oleWindow.GetWindow();
+                    if (root_hwnd != IntPtr.Zero)
+                    {
+                        result.root_hwnd = root_hwnd;
+                        result.is_root_hwnd = true;
+                        return result;
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (!IsExpectedException(e))
+                        throw;
+                }
+            }

```

Comment:
This might lead to incorrect results on older Qt implementations, I believe all of its `IAccessible2` interfaces return an `IOleWindow` implementation that will give the root HWND, even on children.

Reply:
Since I don't have anything available to test, how would you distinguish them?

Reply:
Updated to check that the window it gets is different from the parent.